sdk/communication/chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java

```diff
@@ -0,0 +1,490 @@
+    }
+
+    /**
+     * Adds thread members to a thread. If members already exist, no change occurs.
+     * 
+     * @param chatThreadId Id of the thread to add members to.
+     * @param body Thread members to be added to the thread.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void addChatThreadMembers(String chatThreadId, AddChatThreadMembersOptions body, final Callback<Void> callback) {
+        this.serviceClient.addChatThreadMembers(chatThreadId, body, callback);
+    }
+
+    /**
+     * Remove a member from a thread.
+     * 
+     * @param chatThreadId Thread id to remove the member from.
+     * @param chatMemberId Id of the thread member to remove from the thread.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void removeChatThreadMember(String chatThreadId, String chatMemberId, final Callback<Void> callback) {
+        this.serviceClient.removeChatThreadMember(chatThreadId, chatMemberId, callback);
+    }
+
+    /**
+     * Creates a chat thread.
+     * 
+     * @param body Request payload for creating a chat thread.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void createChatThread(CreateChatThreadOptions body, final Callback<MultiStatusResponse> callback) {
+        this.serviceClient.createChatThread(body, callback);
+    }
+
+    /**
+     * Gets the list of chat threads of a user.
+     * 
+     * @param maxPageSize The maximum number of chat threads returned per page.
+     * @param startTime The earliest point in time to get chat threads up to. The timestamp should be in ISO8601 format: `yyyy-MM-ddTHH:mm:ssZ`.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatThreads(Integer maxPageSize, OffsetDateTime startTime, final Callback<Page<ChatThreadInfo>> callback) {
```

Comment:
I think these should be in an Options class, but I guess that can't be achieved with the autogenerated tool

Reply:
We can achieve anything given enough time. For now, null for optional parameters should do. :-)